pkg/apis/storage/validation/validation.go

```diff
@@ -459,3 +470,56 @@ func validateVolumeLifecycleModes(modes []storage.VolumeLifecycleMode, fldPath *
 
 	return allErrs
 }
+
+// ValidateStorageCapacityName checks that a name is appropriate for a
+// StorageCapacityName object.
+var ValidateStorageCapacityName = apivalidation.ValidatePersistentVolumeName
+
+// ValidateCSIStorageCapacity validates a CSIStorageCapacity.
+func ValidateCSIStorageCapacity(capacity *storage.CSIStorageCapacity) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&capacity.ObjectMeta, false, ValidateStorageCapacityName, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateCSIStorageCapacitySpec(&capacity.Spec, field.NewPath("spec"))...)
+	// There's nothing to validate for CSIStorageCapacityStatus and therefore no validateCSIStorageCapacityStatus.
+	return allErrs
+}
+
+// validateCSIStorageCapacitySpec validates a CSIStorageCapacitySpec.
+func validateCSIStorageCapacitySpec(spec *storage.CSIStorageCapacitySpec, fldPath *field.Path) field.ErrorList {
+	allErrs := apivalidation.ValidateCSIDriverName(spec.DriverName, field.NewPath("driverName"))
+	if spec.NodeTopology != nil {
+		nodeTopoFldPath := field.NewPath("nodeTopology")
+		allErrs = append(allErrs, apivalidation.ValidateNodeSelector(spec.NodeTopology, nodeTopoFldPath)...)
+		allErrs = append(allErrs, validateNoFieldSelector(spec.NodeTopology, nodeTopoFldPath)...)
+	}
+	for _, msg := range apivalidation.ValidateClassName(spec.StorageClassName, false) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("storageClassName"), spec.StorageClassName, msg))
+	}
+	return allErrs
+}
+
+// validateNoFieldSelector ensures that MatchFields is empty in all node selector terms.
+func validateNoFieldSelector(nodeSelector *api.NodeSelector, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	nodeSelectorPath := fldPath.Child("nodeSelectorTerms")
+	for i, term := range nodeSelector.NodeSelectorTerms {
+		if len(term.MatchFields) > 0 {
+			allErrs = append(allErrs, field.Forbidden(nodeSelectorPath.Index(i).Child("matchFields"), "only matching by labels is supported"))
```

Comment:
Something to think about - if you really only want a subset of the other type, it may be easier and cleaner to just copy it to your own scope and keep just the fields you want.  Things like openapi will document these fields that we KNOW don't work.  It's bad UX

Reply:
I won;t require it now, but I would suggest it strongly.

Reply:
I was actually hoping that NodeSelector fields support was more generalized across all of Kubernetes and not specific to each implementation that uses it. I think it would be too big of a change to do here, but it may be worth wrapping the NodeSelector evaluation function in a higher level util that could be used by all components that need to operate on NodeSelectors. There was an attempt to add field support to PV.NodeAffinity that I recall did combine them into a shared util, but unfortunately the PR stalled (and I'm bad at searching and unable to find it now).

Reply:
Silently ignoring fields is pretty bad.  For things that have done this already, it's awkward because it's KIND OF a breaking change.  We should discuss this when moving to beta - please make it a graduation criteria of the KEP?

Reply:
See my other comment here about just using a LabelSelector instead of NodeSelector - does that obviate this?

Reply:
Done.